### PR TITLE
env-vars: add rules and format for console env vars

### LINKED ISSUE #649: Env variable with `.` character
We need to rework this section https://www.clever-cloud.com/doc/develop/env-variables/#variable-definition-constraints

* We should rename it to something "env vars rules and formats" or something...
* We need a sub section to explain what is a valid strict name https://github.com/CleverCloud/clever-client.js/blob/master/esm/utils/env-vars.js#L21
  * having a subsection will allow us to reference it as a link
* We need a sub section to explain the exceptions on names for Java properties https://github.com/CleverCloud/clever-client.js/blob/master/esm/utils/env-vars.js#L32
  * the names with a `.` character
  * having a subsection will allow us to reference it as a link
* We need a sub section to explain the expert mode format
  * We could mention that this format is accepted by our command line
* We need a sub section to explain the json mode format

In another section we could add a note about value encryption https://github.com/CleverCloud/doc.clever-cloud.com/issues/352

### PR DESCRIPTION
**Pull request by Galimede**
Fixes #649

### CODE REVIEW

File: develop/env-variables.md [hunk 1]
```diff
@@ -82,9 +82,131 @@ We use environment variables for some settings:
...
+
+- Empty value:
+```bash
+EMPTY=""
+```
+
+#### Json
+
+The format of a Json variable is: `{ "name": "THE_NAME", "value": "the value" }`. 
+Please note that you must put these variables in an array (see the multiple variable example to see how it goes).
```
Comment (gaelreyrol) [inline develop/env-variables.md:172]:
```suggestion
Note: You must put these variables in an array (see the multiple variable example to see how it goes).
```

**Comment by Galimede** [comment]
Thanks for the review :smile: 
I'm gonna do the changes and provide more realistic examples as well.

**Comment by Galimede** [comment]
I tried to change the examples but I don't know if it's better. WDYT ? 
Otherwise, I don't really know what I should put :thinking:

**Comment by hsablonniere** [comment]
@Galimede I think you should put the examples between single backticks, this way, the ones with underscores won't be in italics and more readable

**Comment by Galimede** [comment]
I've made the modifications is everything OK for you @gaelreyrol, @hsablonniere ?

**Review by gaelreyrol — APPROVED** [review]
LGTM ! Thanks :+1:

**Pull request merged**